flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -109,22 +113,22 @@
 
 	private RestartStrategies.RestartStrategyConfiguration restartStrategyConfiguration;
 	
-	private long taskCancellationIntervalMillis = ConfigConstants.DEFAULT_TASK_CANCELLATION_INTERVAL_MILLIS;
+	private long taskCancellationIntervalMillis = -1;
 
 	// Serializers and types registered with Kryo and the PojoSerializer
 	// we store them in linked maps/sets to ensure they are registered in order in all kryo instances.
 
-	private final LinkedHashMap<Class<?>, SerializableSerializer<?>> registeredTypesWithKryoSerializers = new LinkedHashMap<>();
+	private final UserCodeValue<LinkedHashMap<Class<?>, SerializableSerializer<?>>> registeredTypesWithKryoSerializers = new UserCodeValue<>(new LinkedHashMap<Class<?>, SerializableSerializer<?>>());
 
-	private final LinkedHashMap<Class<?>, Class<? extends Serializer<?>>> registeredTypesWithKryoSerializerClasses = new LinkedHashMap<>();
+	private final UserCodeValue<LinkedHashMap<Class<?>, Class<? extends Serializer<?>>>> registeredTypesWithKryoSerializerClasses = new UserCodeValue<>(new LinkedHashMap<Class<?>, Class<? extends Serializer<?>>>());
 
-	private final LinkedHashMap<Class<?>, SerializableSerializer<?>> defaultKryoSerializers = new LinkedHashMap<>();
+	private final UserCodeValue<LinkedHashMap<Class<?>, SerializableSerializer<?>>> defaultKryoSerializers = new UserCodeValue<>(new LinkedHashMap<Class<?>, SerializableSerializer<?>>());
 
-	private final LinkedHashMap<Class<?>, Class<? extends Serializer<?>>> defaultKryoSerializerClasses = new LinkedHashMap<>();
+	private final UserCodeValue<LinkedHashMap<Class<?>, Class<? extends Serializer<?>>>> defaultKryoSerializerClasses = new UserCodeValue<>(new LinkedHashMap<Class<?>, Class<? extends Serializer<?>>>());
 
-	private final LinkedHashSet<Class<?>> registeredKryoTypes = new LinkedHashSet<>();
+	private final UserCodeValue<LinkedHashSet<Class<?>>> registeredKryoTypes = new UserCodeValue<>(new LinkedHashSet<Class<?>>());
 
-	private final LinkedHashSet<Class<?>> registeredPojoTypes = new LinkedHashSet<>();
+	private final UserCodeValue<LinkedHashSet<Class<?>>> registeredPojoTypes = new UserCodeValue<>(new LinkedHashSet<Class<?>>());
```

Comment:
You forgot to put the `GlobalJobParameters` into a `UserCodeValue`. That one can also contain usercode.